src/storage/lsm.rs

```diff
@@ -160,4 +163,74 @@ where
         }
         Ok(None)
     }
+
+    pub fn get_range<'a>(
+        &'a self,
+        k_lo: &Option<K>,
+        k_hi: &Option<K>,
+    ) -> Result<impl Iterator<Item = Result<(K, V)>> + 'a> {
+        let ssts_iter = SSTable::merge_range(&self.sstables, &k_lo, &k_hi).unwrap();
+
+        let mts_iter = [self.memtable_in_flush.as_ref(), Some(&self.memtable)]
+            .iter()
+            .filter_map(|mt| *mt)
+            .enumerate()
+            .map(|(mt_i, mt)| {
+                let mut iter = mt.iter();
+
+                if let Some(k_lo) = &k_lo {
+                    if let Some(iter_pos) = mt.iter().rposition(|(k, _v)| k < k_lo) {
+                        iter.nth(iter_pos);
+                    }
+                }
+
+                let k_hi = k_hi.clone();
+                iter.take_while(move |(k, _v)| {
+                    // This closure moves k_hi.
+                    if let Some(k_hi) = &k_hi {
+                        k <= &k_hi
+                    } else {
+                        true
+                    }
+                })
+                .zip(std::iter::repeat(mt_i))
+            })
+            .kmerge_by(|((a_k, _a_v), a_i), ((b_k, _b_v), b_i)| {
+                a_k < b_k || (a_k == b_k && a_i > b_i)
+            })
+            .unique_by(|((k, _v), _mt_i)| k.clone())
+            .map(|(kv, _mt_i)| kv);
+
+        let mut ssts_iter = ssts_iter.peekable();
+        let mut mts_iter = mts_iter.peekable();
```

Comment:
The reason we separate out one iter for all SSTables and one iter for both Memtables is that the type of the iterators are different.

- The iterators on SSTables are all `KeyValueIterator`.
- The iterators on Memtables are both `BTreeMap<K, V>::Iter` (or something like that)

To put these two different types into one vec requires `Vec<Box<dyn Iterator<Item = ...>>>`. This makes it cumbersome.

In addition, the iterators' `Item` types are different:

- The iterators on SSTables have `Item = Result<(K, V)>`
- The iterators on Memtables have `Item = Result<&(K, V)>`

For these reasons, it's simpler to keep them separate.